queue.java

```diff
@@ -0,0 +1,76 @@
+//time complexity is o(n)
+//space complexity is o(n)
```

Comment:
Worst case time complexity is O(n) but average case is O(1). Just to make you aware of it